pkg/storage/stores/shipper/indexshipper/tsdb/index/chunk.go

```diff
@@ -105,6 +139,51 @@ func (c ChunkMetas) Finalize() ChunkMetas {
 
 }
 
+// Stats returns the ChunkStats. It tries to accommodate for overlapping chunks.
+func (c ChunkMetas) Stats(from, through int64, deduplicate bool) ChunkStats {
+	res := ChunkStats{}
+
+	if len(c) > 1 && deduplicate {
+		sort.Sort(c)
+		last := c[0]
+		n := 1
+		for _, cur := range c[1:] {
+			// Skip chunk if it's a subset of the last one
+			if cur.MinTime < last.MaxTime && cur.MaxTime <= last.MaxTime {
+				continue
```

Comment:
Would it make sense to take the max(cur, last) instead of skipping? I am a newbie of how chunks are formed, but if one of them could be a subset of the other, nothing guarantees that the one coming first is the "bigger".

Reply:
That's true. We should average the size probably.